libraries/chain/controller.cpp

```diff
@@ -2013,7 +2011,9 @@ struct controller_impl {
                // See comment below about pause-at-block for why `|| conf.num_configured_p2p_peers > 0`
                if (chain_head_is_root || conf.num_configured_p2p_peers > 0) {
                   ilog("applying branch from fork database ending with block: ${id}", ("id", pending_head->id()));
-                  maybe_apply_blocks(forked_callback_t{}, trx_meta_cache_lookup{});
+                  // applies all blocks up to forkdb head from forkdb, shouldn't return incomplete, but if it does loop until complete
+                  while (maybe_apply_blocks(forked_callback_t{}, trx_meta_cache_lookup{}) == controller::apply_blocks_result::incomplete)
+                     ;
```

Comment:
should we `check_shutdown() ` here?

Reply:
If we exit `maybe_apply_blocks` because of shutdown then `maybe_apply_blocks` returns `complete`.